File: lib_common/src/main/java/com/pfl/common/base/BaseFragment.java
package com.pfl.common.base;

import androidx.fragment.app.Fragment;
import com.jojo.design.common_ui.dialog.LoadingDialog;

public class BaseFragment extends Fragment {

    private LoadingDialog mDialog;

    public void showDialog() {
        if(mDialog == null || !mDialog.isShowing()) {
            mDialog = new LoadingDialog(getActivity());
            mDialog.show();
        }
    }

    public void dismissDialog() {
        if (null != mDialog && mDialog.isShowing()) {
            mDialog.dismiss();
        }
    }
}

File: lib_common/src/main/java/com/pfl/common/base/BaseFragment.kt
package com.pfl.common.base

import androidx.fragment.app.Fragment
import com.jojo.design.common_ui.dialog.LoadingDialog

open class BaseFragment : Fragment() {

    private var mDialog: LoadingDialog? = null

    fun showDialog() {
        if (mDialog == null || !mDialog!!.isShowing()) {
            mDialog = LoadingDialog(activity!!)
            mDialog!!.show()
        }
    }

    fun dismissDialog() {
        if (null != mDialog && mDialog!!.isShowing()) {
            mDialog!!.dismiss()
        }
    }
}
